domain/removal/state/model/remoterelation.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package model
+
+import (
+	"context"
+	"time"
+
+	"github.com/canonical/sqlair"
+
+	"github.com/juju/juju/domain/removal"
+	"github.com/juju/juju/internal/errors"
+)
+
+// RemoteRelationExists returns true if a relation exists with the input
+// UUID, and relates a synthetic application
+func (st *State) RemoteRelationExists(ctx context.Context, rUUID string) (bool, error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return false, errors.Capture(err)
+	}
+
+	remoteRelationUUID := entityUUID{UUID: rUUID}
+	existsStmt, err := st.Prepare(`
+SELECT r.uuid AS &entityUUID.uuid
+FROM   relation AS r
+JOIN   relation_endpoint AS re ON r.uuid = re.relation_uuid
+JOIN   application_endpoint AS ae ON re.endpoint_uuid = ae.uuid
+JOIN   application AS a ON ae.application_uuid = a.uuid
+JOIN   charm AS c ON a.charm_uuid = c.uuid
+JOIN   charm_source AS cs ON c.source_id = cs.id
+WHERE  r.uuid = $entityUUID.uuid
+AND    cs.name = "cmr"`, remoteRelationUUID)
```

Comment:
```suggestion
AND    cs.name = 'cmr'`, remoteRelationUUID)
```

https://www.sqlite.org/quirks.html#double_quoted_string_literals_are_accepted